### PR TITLE
Compare example and new conversion utilities

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Added an example for using the comparison code.  While constructing this, found it was best to add a new conversion utility and re-arrange some of the contents of match.  The documents now reflect this change and unit tests were updated.  Partially addresses #35.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- This change requires a documentation update

# How Has This Been Tested?

Added unit tests and created an example that is now in the docs.  Recommend following it to fully test the PR.

**Test Configuration**:
* Operating system: OS X Mojave
* Version number: 3.9
* Any details about your local setup that are relevant: develop branch of pysat

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [N/A] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysatModels/tests/test_utils_convert.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        self.model_inst = pysat.Instrument(**self.model_kwargs)
+        self.xout = convert.load_model_xarray(self.ftime, self.model_inst)
+
+        # Evaluate the returned data
+        eval_xarray_output(self.model_inst, self.xout)
+        return
+
+
+class TestUtilsConvertPysatXarray():
+    """ Unit tests for utils.convert.load_model_xarray
```
Comment (aburrell) [inline pysatModels/tests/test_utils_convert.py:134]:
```suggestion
    """ Unit tests for utils.convert.convert_pydat_to_xarray
```

File: pysatModels/tests/test_utils_convert.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                        else:
+                            raise ValueError('Unknown metadata type')
+    return
+
+
+class TestUtilsConvertLoadModelXarray():
+    """ Unit tests for utils.convert.load_model_xarray
+    """
+    def setup(self):
+        """ Runs before every method to create a clean testing setup
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:68]:
```suggestion
        """Create a clean testing setup before each method.
```

File: pysatModels/tests/test_utils_convert.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        self.filename = "%Y-%m-%d.nofile"
+        self.model_kwargs = {'platform': str('pysat'),
+                             'name': str('testing_xarray'),
+                             'num_samples': 12,
+                             'clean_level': 'clean'}
+        self.model_inst = None
+        self.xout = None
+        self.temp_file = 'None'
+
+    def teardown(self):
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:80]:
```suggestion
    def teardown(self):
        """Clean up test environment after each method."""
```

File: pysatModels/tests/test_utils_convert.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        if os.path.isfile(self.temp_file):
+            os.remove(self.temp_file)
+
+        del self.ftime, self.model_kwargs, self.xout, self.filename
+        del self.temp_file, self.model_inst
+
+    def test_no_inst(self):
+        """ Test failure when no instrument object is provided
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:88]:
```suggestion
        """Test failure when no instrument object is provided.
```

File: pysatModels/tests/test_utils_convert.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        """
+        with pytest.raises(ValueError) as verr:
+            convert.load_model_xarray(self.ftime)
+
+        assert verr.value.args[0].find("must provide a pysat.Instrument") >= 0
+        return
+
+    @pytest.mark.parametrize("fname", [(None), ('filename')])
+    def test_load_filename(self, fname):
+        """ Test success when loading through different filename options
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:98]:
```suggestion
        """Test success when loading through different filename options.
```

File: pysatModels/tests/test_utils_convert.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+                                              filename=fname)
+
+        # Evaluate the returned data
+        eval_xarray_output(self.model_inst, self.xout)
+        return
+
+    @pytest.mark.parametrize("mkey, mval", [("name", "testing"),
+                                            (None, None)])
+    def test_load_inst(self, mkey, mval):
+        """ Test success when loading different types of pysat Instruments
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:121]:
```suggestion
        """Test success when loading different types of pysat Instruments.
```

File: pysatModels/tests/test_utils_convert.py [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        self.model_inst = pysat.Instrument(**self.model_kwargs)
+        self.xout = convert.load_model_xarray(self.ftime, self.model_inst)
+
+        # Evaluate the returned data
+        eval_xarray_output(self.model_inst, self.xout)
+        return
+
+
+class TestUtilsConvertPysatXarray():
+    """ Unit tests for utils.convert.convert_pydat_to_xarray
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:134]:
```suggestion
    """Unit tests for utils.convert.convert_pydat_to_xarray.
```

File: pysatModels/tests/test_utils_convert.py [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+    """
+    def setup(self):
+        """ Runs before every method to create a clean testing setup
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:137]:
```suggestion
        """Create a clean testing setup before each method.
```

File: pysatModels/tests/test_utils_convert.py [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+        """
+        self.ref_time = pysat.instruments.pysat_testing._test_dates['']['']
+
+    def teardown(self):
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:141]:
```suggestion
    def teardown(self):
        """Clean up test environment after each method."""
```

File: pysatModels/tests/test_utils_convert.py [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+        del self.ref_time
+
+    @pytest.mark.parametrize("name, load", [('testing', False),
+                                            ('testing', True),
+                                            ('testmodel', True)])
+    def test_convert_pysat_to_xarray(self, name, load):
+        """ Test success when converting pysat data to an xarray.Dataset
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:148]:
```suggestion
        """Test success when converting pysat data to an xarray.Dataset.
```

File: pysatModels/tests/test_utils_convert.py [hunk 11]
```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019, pysat team
+# Full license can be found in License.md
+# -----------------------------------------------------------------------------
+
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_convert.py:6]:
```suggestion
"""Unit tests for pysatModels.utils.convert."""
```

File: pysatModels/utils/__init__.py [hunk 12]
```diff
@@ -1,19 +1,15 @@
...
 # -*- coding: utf-8 -*-
-# Copyright (C) 2019, AGB & pysat team
+# Copyright (C) 2019, pysat team
 # Full license can be found in License.md
 # -----------------------------------------------------------------------------
 """
-pysatModels.utils
-=================
-
 Utilities designed to extract, match, and compare modelled and observed data
```
Comment (jklenzing) [inline pysatModels/utils/__init__.py:7]:
```suggestion
Utilities designed to extract, match, and compare modelled and observed data.
```

File: pysatModels/utils/compare.py [hunk 13]
```diff
@@ -16,7 +16,8 @@
 import pysatModels as ps_mod
 
 
-def compare_model_and_inst(pairs, inst_name, mod_name, methods=['all']):
+def compare_model_and_inst(pairs, inst_name, mod_name, methods=['all'],
+                           unit_label='units'):
     """Compare modelled and measured data
```
Comment (jklenzing) [inline pysatModels/utils/compare.py:21]:
```suggestion
    """Compare modelled and measured data.
```

File: pysatModels/utils/convert.py [hunk 14]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019, pysat team
+# Full license can be found in License.md
+# -----------------------------------------------------------------------------
+"""
+Routines to support extracting pysat.Instrument data as xarray.Datasets
```
Comment (jklenzing) [inline pysatModels/utils/convert.py:7]:
```suggestion
Routines to support extracting pysat.Instrument data as xarray.Datasets.
```

File: pysatModels/utils/convert.py [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+
+"""
+from os import path
+
+
+def load_model_xarray(ftime, model_inst=None, filename=None):
+    """ Load and extract data from a model Instrument at the specified time
```
Comment (jklenzing) [inline pysatModels/utils/convert.py:14]:
```suggestion
    """Load and extract data from a model Instrument at the specified time.
```

File: pysatModels/utils/convert.py [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+        model_inst.load(date=ftime)
+
+    # Extract the xarray Dataset, returning None if there is no data
+    model_xarray = convert_pysat_to_xarray(model_inst)
+
+    return model_xarray
+
+
+def convert_pysat_to_xarray(inst):
+    """ Extract data from a model Instrument as a Dataset with metadata
```
Comment (jklenzing) [inline pysatModels/utils/convert.py:56]:
```suggestion
    """Extract data from a model Instrument as a Dataset with metadata.
```

**Review by jklenzing — COMMENTED** [review]
Can't test the examples because of #86, but here are some style comments.

File: pysatModels/tests/test_utils_convert.py [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+    Parameters
+    ----------
+    inst : pysat.Instrument
+        pysat Instrument object with or without loaded data
+    xdata : xarray.Dataset or NoneType
+        Data extracted from pysat.Instrument, with metadata if possible
+
+    Raises
+    ------
+    AssertionError
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pysatModels/tests/test_utils_convert.py:29]:
```suggestion
    ValueError
```

**Review by jklenzing — COMMENTED** [review]
The tests look good, but I'm having trouble running them locally.  I keep getting the error `AttributeError: module 'verify' has no attribute 'bias'` when I try to run.

**Comment by aburrell** [comment]
Which version of pyForecastTools do you have?

**Comment by jklenzing** [comment]
> Which version of pyForecastTools do you have?

1.1.1

**Comment by jklenzing** [comment]
It's the same version as on github actions.  Very confused as to why it succeeds there but fails locally.

**Comment by aburrell** [comment]
Ok, I just updated to that version and it ran without any problems.  I also checked the PyForecastTools code and it has a `bias` function.  So... not sure why you're getting that error.

**Comment by jklenzing** [comment]
If I import `verify` directly in ipython, I don't see the bias or other statistic functions in the imported package

**Comment by jklenzing** [comment]
Looking at the code at pyForecastTools, those functions are imported vi `import *`, which can act differently depending on install.  I'm guessing this is the root cause for me.  Going to open an issue over there.

**Comment by jklenzing** [comment]
Update: I've deleted the pip-installed version and installed directly from github.  This is working fine now, so I assume it is a problem with the relative path import and my specific system.

**Comment by aburrell** [comment]
The document test is failing due to UCAR.  Cool to merge upon approval.

**Review by jklenzing — APPROVED** [review]

**Pull request merged**